cosmos/airflow/graph.py

```diff
@@ -171,10 +171,14 @@ def build_airflow_graph(
             node=node,
             execution_mode=execution_mode,
             args=task_args,
-            use_name_as_task_id_prefix=test_behavior != TestBehavior.AFTER_EACH,
+            use_name_as_task_id_prefix=not (test_behavior == TestBehavior.AFTER_EACH and node.has_test),
```

Comment:
It feels we should perhaps rename this argument (`use_name_as_task_id_prefix`) to `should_use_task_group` or something similar since not only is it now affecting the name, but also the graph topology (single node x a task group with two nodes), what do you think?

Reply:
Thank you for the review, i agree that the argument must be changed because for some people the argument can be confusing. I think it's more clearly if we can add a property `should_use_task_group` on TaskMetadata. So we can simplify the branching on the code like this
```py
# Before Change
task_meta = create_task_metadata(...)
if task_meta and node.resource_type != DbtResourceType.TEST:
  if (
      node.resource_type == DbtResourceType.MODEL
      and node.has_test is True
      and test_behavior == TestBehavior.AFTER_EACH
  ):

# After change
task_meta = create_task_metadata(...)
if task_meta.should_use_task_group is True:
```

Reply:
Resolved on the newest commit